### DISCUSSION TITLE
A case for bartering in multiplayer games

### DISCUSSION THREAD

**Discussion by Griatch** [evennia/evennia#2347] [General Discussion]
_This is a repost of an artice I posted for the Imaginary Realities e-zine, volume 7, issue 2 back in 2015. Since IR is no longer active I repost it here with some cleanup._

# A Case For Bartering In Multiplayer Games

_by Griatch, 19th February 2015_
_images from https://deviantart.com/griatch-art_

![Troll deal](https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/22916c25-6299-453d-a221-446ec839f567/d1b6v6j-3ab96f99-d8cc-482c-9303-4239aa672566.jpg/v1/fill/w_300,h_425,q_75,strp/troll_deal_by_griatch_art_d1b6v6j-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOiIsImlzcyI6InVybjphcHA6Iiwib2JqIjpbW3siaGVpZ2h0IjoiPD04NDkiLCJwYXRoIjoiXC9mXC8yMjkxNmMyNS02Mjk5LTQ1M2QtYTIyMS00NDZlYzgzOWY1NjdcL2QxYjZ2NmotM2FiOTZmOTktZDhjYy00ODJjLTkzMDMtNDIzOWFhNjcyNTY2LmpwZyIsIndpZHRoIjoiPD02MDAifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6aW1hZ2Uub3BlcmF0aW9ucyJdfQ.5lrvvyoPc4u8oLig0fLvHBGVxqL8V0Y9FipjEdViDt0)
_Troll deal_

Most multiplayer online games eventually introduce some sort of economic system. Maybe there are NPC vendors to sell to, maybe there is an auction house or just the ability to trade with one another. An economic system of some description is expected in most games, if only for offloading your loot or upgrading your weapons. In any sort of consistent-world RPG it is also realistic — it makes sense that the in-game metropolis should have a burgeoning marketplace.

From a game designer perspective, though, in-game economy has only one actual purpose: to produce emergent gameplay. Emergent gameplay roughly means that your players create their own game enjoyment out of the building blocks you supply - potentially extending the game beyond the sum of its parts.

Economy is a great candidate for emergent gameplay. Players are inherently familiar with the concept of trade. By even introducing the possibility to buy and sell, you give players the ability to create their own motivations and plots. It gives them stories of how they finally gathered enough money to buy that upgrade or managed to corner the market on rat tails. Fun all around ...!

… In theory, anyway. In reality, game economy is tricky. In most games it's not really an 'economy' at all, but just a series of vending machines (named NPC shop keepers) with fixed prices. Here I'll go a little deeper than that. The problem with a real economy is that for it to be useful and fun, players must perceive it as completely fair yet rewarding to manipulate. This article will try to argue against a single-currency money-based economy and try to make a case for using a multi-currency bartering economy instead.

## Thought experiment: The peaceful NPC economy

![Troll gifts](https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/22916c25-6299-453d-a221-446ec839f567/d2ehe10-78eca7a7-7f60-4c9f-918a-891167d774d4.jpg/v1/fill/w_500,h_367,q_75,strp/christmas_card_2009_by_griatch_art_d2ehe10-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOiIsImlzcyI6InVybjphcHA6Iiwib2JqIjpbW3siaGVpZ2h0IjoiPD02MzciLCJwYXRoIjoiXC9mXC8yMjkxNmMyNS02Mjk5LTQ1M2QtYTIyMS00NDZlYzgzOWY1NjdcL2QyZWhlMTAtNzhlY2E3YTctN2Y2MC00YzlmLTkxOGEtODkxMTY3ZDc3NGQ0LmpwZyIsIndpZHRoIjoiPD05MDAifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6aW1hZ2Uub3BlcmF0aW9ucyJdfQ.GESZOCx95cDOJ3ikCY6YrClTqu4mrUtWmX-A_nN6cfI)
_Gift of trolls_

Consider an ideal money-based game economy.

This economy is a simple closed market only populated by computer-controlled NPCs. It has a fixed total number of money units -  let’s call it “gold” for the sake of this article. 

In our simulation we use some sort of AI to make sure NPCs have _needs_ and can produce _resources_. Whenever an NPC buys something from another NPC, money changes hands but the _total amount of gold in the world does not change_. 
It’s not too hard to emulate supply and demand: just track the global number of goods you have and (with some suitable algorithm) scale up the price for rare goods while scaling down the price for common ones. The system need not be completely deterministic, either; it can have some random ups and downs in goods availability. This is coupled to an artificial delay before a price adjusts to the “real” supply and demand, allowing NPCs to earn extra before prices drop or lose out if they happen to sell stuff at the wrong time.

This is a relatively simple system to code and emulates the overarching principles of a _closed market system_; The NPCs trade their goods for gold, then spend that gold again for things they need. Some benefit from the random variations and delays and get richer than others, others would become poor. This is a place of well-behaved NPCs. If you tweaked the numbers right you should realistically be able to balance and run this simulation indefinitely.

## The players come and ruin everything

Now introduce human players into this peaceful NPC economy and things tend to go haywire. Why is that? Here are only some of the reasons:

- Players join the game from _nowhere_ as far as the game-world ins concerned. Yet they expect to be able to participate as if they were always there. The real-world equivalent might be foreigners moving into a very isolated country, bringing new wealth with them and expecting everyone to just trust their money is good from the get-go.
-  To add that new wealth (even if just a starting wealth) we can't take that from existing players (at least not without a lot of complaints). So the economy must _grow_, meaning the total amount of gold (and goods, but usually only gold is considered) must increase. We are essentially creating new gold from nothing. 
- Conversely, a player _leaving_ the game might or might not return to the game again. This has no real-world equivalent. The character of a logged-out player usually has no in-game needs! The money they own is locked away from the economy. You usually don’t want to erase/reclaim their resources in case absent players decide to come back later. This causes a dilemma: Should the money they carry no longer count towards the total amount of gold? This effect may be minor in a large game like EVE Online, but in a MUD with some 20-40 regular players, every player’s purse matters.
- A player expects to have _fun_. Or at least to have some real influence over their own fate. This has no real-world equivalent either. In the real world you just have to live with a crashing economy or being stuck in a financial situation you can’t get out of. In a game, you can decide to log off and not come back.
- For the sake of ensuring “fun”, players tend to get unfair advantages compared to NPCs. They have access to goods/loot the NPCs cannot get and often travel much faster and over greater distances (this last part is often overlooked - if movement is too fast, the real-world purpose of trade routes falls away and distance ceases to be a factor for rarity. That rare commodity is not so rare if everyone can just teleport directly to the distant place it's made and buy it from the source.
- Many games introduce new gold into the system via monster kills and quest rewards. This is _new_ gold. No one ever traded that gold or used it to buy existing goods. Again, this is used as an incentive to keep the players entertained but it is death to our ideal little economy simulation.
- Finally, human players can’t be expected to show restraint the way NPCs can be coded to do. Our nice supply-and-demand algorithm can easily be subverted if one player decides to spend a week cutting down all the trees in your game to get a monopoly on wood. It can be very hard to dampen the effects of such extreme actions.

Note that from among these points, the irrational human nature is only mentioned in the last one (you must always expect a bored player to test the unexpected just for laughs). The rest stem from the fact that we not only want new players to join our game, we also want them to enjoy their experience.

The unfortunate result of all this is that our carefully designed economy becomes more and more meaningless with time. Rapid _inflation_ kicks in: as more and more gold is introduced into the system, the actual value of each unit drops. And this is only the ideal case: Few games even bother starting from that balanced and friendly NPC economy - they just open to human players and hope for the best. Most give new players a chunk of money and mint new coin for them whenever needed. NPCs are only there to dispense even more money, often at fixed prices. Before long, players bathe in money, making the gold economy all but superfluous except to the absolute newcomers. That impressive sword you worked so hard to afford a few months back? Now it’s worth no more than what you regularly carry in your pocket.

But there is still a need to trade! People will still  have stuff you want. At this point unofficial bartering emerges between players instead, eschewing NPC vendors and gold except for basic things they can’t get otherwise. Trade becomes item-to-item or using multiple _fiat currencies_ - maybe the best magical potions can only be made by player crafters spending all their experience points in that. Then those potions are suddenly a limited resource and will start acting as 'coin'. This is a phenomenon observed on real-world MMORPGs much bigger than your common MU*. 

There are many ways proposed to solve the problems of in-game money economies (short of cheating and alienating your players by removing all their wealth by some hand-wavy fiat). Most do so via _sinks_. Sinks are basically black holes to throw gold into. This could be expensive one-use potions that people need to survive, taxes on transactions or other expenses that effectively removes gold from the system. In theory, if there are as many sinks as there are sources of money you would have an inflation-free economy where things retain their value with time.

However, this risks going against the goal of _fun_. Most players _loathe_ being forced to spend money (taxes are no more fun in-game than they are in the real world). Certain game genres, like PvP-heavy games, can make this more natural (like in EVE online, the major money sink is losing your ship so you have to buy a new one). There are also other ways to 'trick' players into wanting to spend money regularly (for example just calling it 'pay to get a bonus' rather than 'pay to not loose your bonus' makes a big difference psychologically). But adding sinks in a fun way nevertheless puts a lot of effort on the designer and potentially affects the style of game you end up having.

It’s _probably_ possible to make a stable money economy ... but it’s safe to say that most games with a traditional money economy will experience rampant inflation eventually. Most don't even try.

That “easy way to have players create their own content” doesn’t sound quite so easy anymore, does it?

## Go for bartering right away

![here, have a flower!](https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/22916c25-6299-453d-a221-446ec839f567/d7jnc3j-8ca00d23-df0e-4ef9-872e-58fadc047844.jpg/v1/fill/w_450,h_450,q_75,strp/mother_s_day_flower_2014_by_griatch_art_d7jnc3j-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOiIsImlzcyI6InVybjphcHA6Iiwib2JqIjpbW3siaGVpZ2h0IjoiPD05MDAiLCJwYXRoIjoiXC9mXC8yMjkxNmMyNS02Mjk5LTQ1M2QtYTIyMS00NDZlYzgzOWY1NjdcL2Q3am5jM2otOGNhMDBkMjMtZGYwZS00ZWY5LTg3MmUtNThmYWRjMDQ3ODQ0LmpwZyIsIndpZHRoIjoiPD05MDAifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6aW1hZ2Uub3BlcmF0aW9ucyJdfQ.fwsFVUEqqjloFRhvPAkpNe1NYDAGTjGrNkwR3yG6rms)
_Here, have a flower!_

A prospect I think could be explored further in multiplayer games would be to forego the single-currency money system completely and go for the multiple-currency barter economy out of the box. This is not a new idea (Richard Bartle argues this point already in his seminal book _Designing Virtual Worlds_), but it bears repeating: once your gold is rendered worthless by inflation, people will resort to bartering anyway, so why not make the barter system a robust one from the onset?

The simplest way to barter is to simply agree to a trade and then use the game’s **give** command to do the exchange. There are some good reasons to add a more formalized barter mechanic to your game, though:

- Having a coded mechanic makes bartering an “official” part of the game in the mind of the players. It tells them that this is the intended way to do it.
- A coded barter system can make sure that no one holds both goods at the same time, avoiding real-life scammers — this could be a real problem if PvP is otherwise limited. If an in-game thief wants to pull a scam or take off with the goods, this should probably be tied to an actual game skill instead.
- A formalized barter system is easier to adopt for use with NPC traders using the same commands.
- A money economy is actually just a subset of a barter economy. The only difference is that one side trades coins rather than goods. So even if you do choose to have a money economy, a robust barter system allows you to trade everything seamlessly.

Some years back I added such a safe [barter system](https://github.com/evennia/evennia/blob/master/evennia/contrib/barter.py) to Evennia’s optional `contrib/` folder. A MUD session in this system can look like this when player Ada wants to trade with player Borek:

> **Ada>** trade Borek: Hi, I have a nice extra sword. Wanna trade?
> **Borek<** Ada says, “Hi, I have a nice extra sword. Wanna trade?” [Ada wants to open trade with you. Enter ‘trade Ada’ to accept]

As seen, these commands all support an optional additional say/emote for roleplaying (see the last issue of Imaginary Realities where I discussed this idea of extending normal commands with emotes).

>  **Borek>** trade Ada: Hm, I could use a sword … if the price is right.
>  **Borek<** [You are now trading with Ada. Use ‘help trade’ for aid.]
>  **Ada<** Borek says, “Hm, I could use a sword … if the price is right.” [You are now trading with Borek. Use ‘help trade’ for aid.]

At this point, Ada and Borek are in “trading mode”. This means that a subset of trading commands are available to let them offer up items for trade. Should either of them decide to move away (or, say, attack each other), the trade is aborted. No actual objects change hands yet. The system also makes sure that the players actually carry the items they are offering.

> **Ada>** offer sword: This is a nice sword. I need some healing potions. Got any?
> **Borek<** Ada says, “This is a nice sword. I need some healing potions. Got any?” [Ada offers Sharp sword. Use ‘inspect` to examine it]
> **Borek>** inspect sharp sword
> **Borek<** (sees stats and description of the Sharp sword)
> **Borek>** offer 2 healing potion: Nice. Here are two potions for your sword.
> **Ada<** Borek says, “Nice. Here are two potions for your sword.” [Borek offers 2 potion of healing, use ‘inspect’ to examine them]

Ada does not think two healing potions are enough for the sharp sword. She does not have to make a counter-offer every time though; the normal emote/say commands work too:

> **Ada>** say Hey, this is a nice sword. I need something more. Maybe some food?
> **Borek** Ada says, “Hey, this is a nice sword. I need something more. Maybe some food?”
> **Borek>** offer 2 healing potion, 3 rations: Alright, I’ll throw in some quality rations, that’s my final offer.
> **Ada<** Borek says, “Alright. I’ll throw in some quality rations, that’s my final offer.” [Borek offers 2 potion of healing, 3 big rations. Use ‘inspect’ to examine them]

At this point the offer is one sword for two healing potions and a few rations. The players have commands to list the current offering and continue bartering. If the deal is off, they can decline to abandon the trade entirely. To seal the deal both sides must accept the same standing deal:

> **Ada>** accept: You are killing me here, but alright.
> **Borek<** Ada says, “You are killing me here, but alright.” [Ada accepts the offer. Use ‘accept’ or change the offer to continue bartering]

If Borek would change his offer now, Ada’s registered acceptance resets so that she has to re-enter it again. Borek is happy with the deal, though:

> **Borek>** accept: Excellent. Nice making business with you!
> **Borek<** [you both accepted. Deal is made and goods change hands]
> **Ada<** Borek says, “Excellent. Nice making business with you!” [you both accepted. Deal is made and goods change hands]

At this point the barter mode is exited and normal gameplay recommences. The promised items are automatically transferred between the players’ inventories. At no point does a single party have possession of both parts of the trade.

## Drawbacks of bartering

In a bartering system the single currency of money is replaced with multiple currencies — the goods themselves.

The drawbacks of in-game bartering are the same as the drawbacks in the real world: What if you don’t have what the other person wants? A barter economy is very inefficient in larger societies as people specialize in wares that not everyone wants all the time. Money exists for a reason — it greases the wheels of commerce. This means that you must design the game so that players require a wide range of goods to be successful at what they do. That is, you must create _supply and demand_. All items in the game must be useful in some way so as to give them complementary trade value. A crafting system is a natural consequence. 
The goods need not be equally diverse for all games, though. One could picture a combat-oriented game where only weapons and armor are tradable. For this to work one would have to design all weapons having their pros and cons and none being universally better than the rest … come to think of it, that sounds like good design advice overall!

And you don't have to get rid of money completely. Just allow one side of the trade be 'coins' and you put gold back into the equation. Question is only if the other side values those coins high enough to accept the trade.

In small games with limited user bases the limits of bartering actually offer some interesting gaming and roleplaying possibilities. It gives players with desirable goods the incentive to set their price: goods that other Players need to procure for them — that is emergent gameplay right there. 
This can help tie the economy more closely to non-violent game mechanics like crafting, potentially making the latter more diverse and rewarding. Combat still has a role (where else to get wolf pelts and lizard hides?) but the spoils of battle are not as generally useful if you don’t have the right buyers lined up, just as in the real world.

## Conclusions

![The human pit stop](https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/22916c25-6299-453d-a221-446ec839f567/d11munx-4e9c01ed-1216-46ec-9288-dcd20901f967.jpg/v1/fill/w_550,h_417,q_75,strp/the_human_pit_stop_by_griatch_art_d11munx-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOiIsImlzcyI6InVybjphcHA6Iiwib2JqIjpbW3siaGVpZ2h0IjoiPD02MzYiLCJwYXRoIjoiXC9mXC8yMjkxNmMyNS02Mjk5LTQ1M2QtYTIyMS00NDZlYzgzOWY1NjdcL2QxMW11bngtNGU5YzAxZWQtMTIxNi00NmVjLTkyODgtZGNkMjA5MDFmOTY3LmpwZyIsIndpZHRoIjoiPD05MDAifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6aW1hZ2Uub3BlcmF0aW9ucyJdfQ.ChQfsZquZD_XbK9pUFHOjSBJbFOzi7-g81OYXiWK5e0)
_The human pit stop_

An economic system is not something you should just put into your game as an afterthought because it “needs to be there”. There are great gameplay benefits to be had from an economy more complex than “throw gold at the players until they stop complaining”

Although this article argues for the benefit of a barter (multi-currency) economy, a money (single currency) economy is probably easier to create. Tracking all these currencies is more work than counting gold. But it also means that the currency of your world is spread more broadly — it’s harder to crash or inflate all those currencies at the same time. You can be the sole supplier of lizard livers but be low on snake tongues ... 

... And any day now those crafty witches will figure out that snake tongues work just as well as lizard livers in their potions.

**Comment by TehFamine** [comment]
> However, this risks going against the goal of fun.

I do feel this is a bit bias. EVE Online, a game I've played since beta religiously, does things well because loss happens, and loss can be fun. Trading for example, someone who spends time in this so-called economy as their profession, has wins and losses. Sometimes they churn a profit, sometimes they take a loss. This is the balance of the system when it comes to the in-game economy.

To say, that loss is a risk of the goal of fun is a misunderstanding of why we as players want these systems. We cannot have an in-game economy without loss. You have to create a demand. That demand comes from a need. The need is often in result of either growth (e.g.: I need more swords because I gained 10,000 more soldiers) or loss (e.g.: I need more swords because I lost 10,000 more soldiers).

I feel that in order for a good in-game economy to thrive, one needs to consider that loss and how to make it fun to lose. Otherwise, what exactly are we bartering for if we already have everything we need because we never lose anything? Logically, we are only gaining better and more powerful items (e.g.: linear progression) until we hit a ceiling. Then what?

**Reply by VRsutzBro** [reply]
have played many VR Games so far now developing the intrest in developing a [réalité virtuelle](https://www.vrsutz.com/fr/realite-virtuelle) game my self

**Comment by Griatch** [comment]
@TehFamine Ah, but that's not the _goal of fun_ intended above. Having losses due to you guessing wrong in the market or selling at the wrong time is awesome. 

_Sinks_, as referenced in the text are artificial things though, imposed by the game. These can be things outside of the agency of the player, which is the problem (a tax is the example in the text). In the next part of that paragraph I mention that such sinks need to be properly presented so as to be acceptable.

**Reply by TehFamine** [reply]
> @TehFamine Ah, but that's not the _goal of fun_ intended above. Having losses due to you guessing wrong in the market or selling at the wrong time is awesome.
> 
> _Sinks_, as referenced in the text are artificial things though, imposed by the game. These can be things outside of the agency of the player, which is the problem (a tax is the example in the text). In the next part of that paragraph I mention that such sinks need to be properly presented so as to be acceptable.

Yeah, but these sinks are not what you add to a proper in-game economy. Sinks you're referring to are things developers add to the game when they don't really have a in-game economy. Not the same thing I'm referring to at all.

For example, using EVE for reference, sinks are tied to the market. They are not generated by the developers, but the players. Everything you buy (or spend gold on) should come from the hands of the player, not something artificial like a NPC with the exception of monster drops. But, those are treated like resources such as wood, metal, ore, etc. Players don't grow the trees. They just harvest them. The same is true from special items that drop from NPC's that are not made by players.

At the end of the day. You put tools in place for players to create their own sinks. They create those sinks through hard loss. If you don't have hard loss, the demand is minimal because over time, players acquire everything they need to reach end game. Once they hit that end game. Why use the economy at that point? Potions? 

Like the reason they buy potions, they consume them and they are gone. Armor, weapons, whatever all need to be gone at some point from the players lifecycle. This either can come from ware and tare or what I would suggest, a bloodbath. :)

**Reply by Griatch** [reply]
The main player-driven sink in EVE is the blowing up of spaceships as far as I remember. It is a very special style of game play that doesn't fit everywhere. They have consumables but I don't recall if they have wear&tear or other artificial sinks in the economy. At any rate they are still heavily affected by inflation. 

Nevertheless, EVE's economy works and feels 'fair' because of its size. If you aim for a fully player-driven economy, you will still need to bootstrap it with artificial sinks when you start up - simply because there will be too few players to keep it rolling. On the smaller scale you need to be much more careful about imbalancing your economy. 

That said - I agree that if you _do_ manage to add your sinks as part of the game in a way that makes your economy roughly balanced and where  players enjoy it while retaining a feel of challenge and making winning worth something, that's _great_. We want more of such games. I personally know of no such MU* game today, but maybe there are some.

It doesn't change the fact that many/most games will never get to such a delicate game economy balance. They may not even be interested in spending the effort of such a thing. Thus I suggest that including a coded barter mechanism right away may be prudent.

**Reply by TehFamine** [reply]
> you will still need to bootstrap it with artificial sinks when you start up

Honestly, I don't think you do. If you make a game where all items can be bought and sold through orders as well make majority of all items craftable by players, then that's enough sinks. You will still have the two types of players. Those that will take time to harvest and those that won't. Those who choose to harvest, will either sell the raw resources on the market or build something with it and sell the final product on the market. Those who don't, will either buy the raw resources to build themselves or buy the final product. In either case, someone is sinking time and money into the result.

If you take the common incorrect way of doing things like making a crafting system where the crafting items are not as good as 90% of the other items in the game, then yeah, you're going to be stuck. You will only further enhance that stuck with not making items, resources, etc part of some type of hard loss. 

I don't care what type of flavor you like. You need loss. You cannot have a market without a demand, especially in a smaller player base. You need to make a demand like a haircut. Maybe not every day, but every couple weeks or something. Keep players coming back to the market. In EVE, it's more like food because people loose ships every morning, lunch, and dinner. And really, it's not the end of the world for these players either.

Harvesting, building, crafting, etc all feed into a healthy economy as well healthy PvP system, that in return, enhances the PvE system by creating that demand on guess what? Harvesting, building, crafting all over again. :)

**Reply by Griatch** [reply]
>> you will still need to bootstrap it with artificial sinks when you start up

> Honestly, I don't think you do. If you make a game where all items can be bought and sold through orders as well make majority of all items craftable by players, then that's enough sinks. 

That players can trade everything is a good thing, agreed. It also strongly supports the case of wanting to have bartering to complement money transactions (for maximum economic flexibility if nothing else). Your sentence there suggest no sinks though. People buying things from each other just means money changing hands, not wealth getting removed. 

Except for the destruction of spaceships, all the sinks you mention (haircuts, food) are very 'artificial' sinks not really a part of the main game loop. Eating food and needing a haircut easily become chores (the kind of things people use scripting to automate, which means they should probably be automated by the game instead). The degradation of weapons and armor is also a very arbitrary thing (there's also a contingent of users that really hate weapon degradation, but that's a separate issue). I don't argue that loss (=sinks) are not needed, I argue that figuring out how to present these artificial things to the player in a way that makes it interesting and not anti-fun is the challenge here.

**Reply by TehFamine** [reply]
True, but fun is subjective here. I should rather say that if you plan to do in-game economy a game with no hard loss, you will find it extremely hard to pull off because over time, for each player, their demand for buying items off the market or bartering for them will decrease and decrease while the demand for outfitting/selling items to new players (or new characters from old players) will increase.

Good in-game economies thrive on blood. Why do you think nations really want to war all the time? 🗡️

**Reply by Griatch** [reply]
I think we are in agreement here - for the 'ideal' comprehensive game economy. If loss from war/conflict is enough of a natural sink in your game (and players accept this), that's great. It's not enough for EVE - they have had rampant inflation since the beginning. 
The thing is though, very few games (certainly not MUDs) even _attempt_ to make such a balanced economy, even less something as comprehensive as that of EVE. On this scale, bartering support remains something I think people should consider.

(I still think one does need a lot of hidden mechanisms under the hood to dampen a virtual economy and lots of artificial sinks to curb inflation while the number of players is small ... but maybe that's a different discussion.)

**Comment by Griatch** [comment]
> I feel that in order for a good in-game economy to thrive, one needs to consider that loss and how to make it fun to lose. Otherwise, what exactly are we bartering for if we already have everything we need because we never lose anything? Logically, we are only gaining better and more powerful items (e.g.: linear progression) until we hit a ceiling. Then what?

This is a good point. The common ways to do this though - perma-death, item degradation etc - do tend to not be very popular with players, is my impression. Most _do_ consider their character development to be as one-way street. How would one make the player lose but still give them the proper feeling of agency?

**Comment by InspectorCaracal** [comment]
This is a really interesting post/conversation on one of my favorite topics! There's a lot of difficulties in setting up a barter economy that the OP touches on, and the discussion about money _sinks_ covers most of the important points there, but I want to look at a different question that's commonly glossed over.

How _and why_ do players get money in the first place?

In RPGs or RPG-based games in particular (which MUDs generally are), the reason for constant inflation is very simple, and mentioned in the OP. Money and resources are constantly being added to the game. But why? Well, because money is treated as a reward for playing the game, hand in hand with experience. You fight a monster, you defeat it, you get experience and loot. But there's no "level cap" on money, unlike experience, so the influx of new resources is infinite. Thus, endless inflation.

Most games don't get past that idea of resources/money being doled out from the aether like experience points, which is why there's so much of a focus on _sinks._ But what if, instead of adding more and more sinks, you changed the rate and source of resources and money being _added_ to the economy in the first place?

In other words: what if the things that give you experience _don't_ give you money, and vice versa? Take the standard Adventurer's Guild concept as a start point, in a game focused on fighting things and improving your ability to fight things. There's a board where people post quest bounties of various difficulty levels. Whether or not you can take and complete one of those bounties depends on your character's abilities, which you improve by getting experience. You get experience by fighting things and winning. But if you want to get _paid,_ you have to pick up a quest, complete it, and return to pick up your rewards. The game can then tune how often to post bounties of what level difficulties and what size rewards to balance the flow of money into the game.

On its own, that example doesn't solve the problem entirely, because you still are creating money out of nothing. But it does lower the need for sinks to a manageable level that can be handled just with a standard approach like repairing gear and medical supplies (usually health potions) while still sticking fairly close to a standard RPG design structure.

The thing about this question - where is the money coming from and why? - is that it would affect a barter economy just as much as a currency economy, because it applies just as much - if not more so - to where all of the resource in the game come from.

Where do the crafting materials come from, and how do you get them?
Where do the final-result items that people want (be it aesthetic clothing or powerful gear) come from, and how do you get them?

And so on. The really important part there for balancing an economy is the "how do you get them" question. Because the resources and materials are always going to be coming out of nowhere, unless you have a completely static world inventory, and "how do you get them" is the best place to put your sinks. Most of the time, "how do you get them" is answered with "by fighting monsters", so the sinks go onto fighting monsters - needing health potions, armor being damaged, scaling experience down so you need to spend money on buff potions to progress, etc. - but it could be on anything.

In short, the key is to make sure that players have to spend resources in order to earn resources.

**Reply by Griatch** [reply]
It's not a bad thought to limit the hand-out of money (give XP instead of coin and vice versa). The counter-point of that is as always the player's 'fun' - players enjoy getting rewards. A game feeling 'stingy' about its rewards can instead make it less worthwhile to do things. It's a delicate balance, since the players don't care about your game economy - they _do_ care about getting properly rewarded when they do something good. I'm not saying it's wrong to limit rewards in order to have fewer sinks, but it seems most modern games find it more worthwhile to give players plenty of immediate satisfaction and then just deal with the consequences via sinks (or don't deal at all, accepting hyper-inflation). 

I agree that balancing a barter economy is itself a quite daunting task - one needs to link this (most likely) to crafting and into the main gameplay loop itself. "Spending resources to fight resources" is also a good thing from our perspective as game devs. Yet things like equipment-wear is often one of the things players really, really dislike in a game. Realistic or not - being able to lose something you fought hard to win is just not 'fun' for many. For it to be worthwhile, the game must be built around the concept so it doesn't feel tacked on and arbitrary.

**Comment by blongden** [comment]
All really good points. Inflation can happen all over the place too, trees can overproduce fruit (so you need to remove it from the game effectively through rotting). All commodity items - things used for trade or to make other items - need to be monitored to make sure that they are scarce enough or abundant enough to support the player economy. This is REALLY hard to do before you have players, so starting out with a bartering system and introducing currency at a later point would make loads of sense.

It becomes even more interesting when commodities are actually difficult to move around. Silver and Gold might be valuable, but they're heavy and transporting them may become slow and difficult (and high risk!) when making larger transactions. Tying in game currency to a volume of rare commodity that has already built up MAY then become a good idea (but completely rules out the option of 'printing' new coins for thing like quest rewards, without devaluing the commodity it's tied to...) 

I have implemented currency in my own world as the currency used for shops, however having NPC's barter for items that they sell makes a lot of sense until we actually need another way of representing value. This after all, is why money exists in the first place.

**Reply by Griatch** [reply]
> It becomes even more interesting when commodities are actually difficult to move around. 

This, I think, is a very tricky thing to implement in a game. On one hand you may want resources to be found only in a given location - this is good because it automatically makes them more rare and gives gameplay opportunities for retrieving/trading. But on the other hand, you may want users to have quick-jump/teleport abilities to quickly move around the world. Infact, the latter is often considered a basic necessity for a big world. But while this makes players happy, it also breaks your economy since distances cease to matter. Being able to carry less can help, but a 'realistic' carry capacity is often not very 'fun' for a user ... It's a balance between the needs of the game economy and presenting it to a user in a way that feels consistent and part of the game for them. If the game is about gritty hyper-realism then they may accept that. Less so in a WoW-like high fantasy setting.

**Reply by blongden** [reply]
Distances only matter when you are transporting commodities (let's say you can only carry small amounts, a handful or what you can fit in a backpack). For larger quantities you would have to load them into a cart or wagon, and transport them by road using a horse and cart (either driven by yourself or hire an NPC to take them). It would take time to move, but maintains the players ability to move quickly.

Of course, I'm just thinking out loud at this point. I'll need to try it out.
